### PR TITLE
Cuda 11.1 cudart SO name fix

### PR DESCRIPTION
**Pull request by nluehr**
In CUDA 11.1 the soname of libcudart remains versioned as 11.0 for better backward comparability. This PR updates the configure scripts accordingly.

This should also address issue #43689 

Attn @sanjoy

### CODE REVIEW

File: third_party/gpus/cuda_configure.bzl [hunk 1]
```diff
@@ -668,6 +669,10 @@ def _get_cuda_config(repository_ctx, find_cuda_config_script):
     cudnn_version = ("64_%s" if is_windows else "%s") % config["cudnn_version"]
 
     if int(cuda_major) >= 11:
+        if int(cuda_major) == 11:
+          cudart_version = "64_110" if is_windows else "11.0"
```
Comment (sanjoy) [inline third_party/gpus/cuda_configure.bzl:673]: Can you please add a comment here on why we need this special logic (same text as the PR description is fine).

**Review by sanjoy — CHANGES_REQUESTED** [review]
LGTM with a minor nit.

File: third_party/gpus/cuda_configure.bzl (hunk 1 above)
Comment (nluehr) [reply third_party/gpus/cuda_configure.bzl:673]: Done.

**Review by sanjoy — APPROVED** [review]

**Comment by jacobaustin123** [comment]
This is currently preventing TF and Jax builds on 11.1 (at least with Ampere cards). Making these changes to the bazel config fixes the issue.

**Pull request merged**